tests/eas/generic.py

```diff
@@ -0,0 +1,450 @@
+                            'period_ms': WORKLOAD_PERIOD_MS,
+                        },
+                        'tasks' : 2,
+                    },
+                    'small' : {
+                        'kind' : 'Periodic',
+                        'params' : {
+                            'duty_cycle_pct': 10,
+                            'duration_s': WORKLOAD_DURATION_S,
+                            'period_ms': WORKLOAD_PERIOD_MS,
+                        },
+                        'tasks' : 3,
+                    },
+                },
+            },
+        },
+    }
+    @experiment_test
+    def test_slack(self, experiment, tasks):
+        self._test_slack(experiment, tasks)
+    @experiment_test
+    def test_task_placement(self, experiment, tasks):
+        self._test_task_placement(experiment, tasks)
+
+class RampUp(EnergyModelTest):
+    """
+    Test EAS for a task ramping from 5% up to 70% over 2 seconds
+    """
+    workloads = {
+        "ramp_up" : {
+            "type": "rt-app",
+            "conf" : {
+                "class"  : "profile",
+                "params"  : {
+                    "r5_10-60" : {
+                        "kind"   : "Ramp",
+                        "params" : {
+                            "period_ms" : 16,
+                            "start_pct" :  5,
+                            "end_pct"   : 70,
+                            "delta_pct" :  5,
+                            "time_s"    :  2,
+                         },
+                    },
+                },
+            },
+        },
+    }
+
+    @experiment_test
+    def test_slack(self, experiment, tasks):
+        self._test_slack(experiment, tasks)
+    @experiment_test
+    def test_task_placement(self, experiment, tasks):
+        self._test_task_placement(experiment, tasks)
+
+class RampDown(EnergyModelTest):
+    """
+    Test EAS for a task ramping from 70% down to 5% over 2 seconds
+    """
+    workloads = {
+        "ramp_down" : {
+            "type": "rt-app",
+            "conf" : {
+                "class"  : "profile",
+                "params"  : {
+                    "r5_10-60" : {
+                        "kind"   : "Ramp",
+                        "params" : {
+                            "period_ms" : 16,
+                            "start_pct" : 70,
+                            "end_pct"   :  5,
+                            "delta_pct" :  5,
+                            "time_s"    :  2,
+                         },
+                    },
+                },
+            },
+        },
+    }
+
+    @experiment_test
+    def test_slack(self, experiment, tasks):
+        self._test_slack(experiment, tasks)
+    @experiment_test
+    def test_task_placement(self, experiment, tasks):
+        self._test_task_placement(experiment, tasks)
+
+class EnergyModelWakeMigration(EnergyModelTest):
+    """
+    Test EAS for tasks alternating beetween 10% and 50%
```

Comment:
"Test EAS for tasks alternating beetween 10% and 50% over 4 seconds". It would be good to specify the duration of the entire workload as for the time axis of the plot to be better understood. The same for the other classes. Some of them include this information but some of them do not.